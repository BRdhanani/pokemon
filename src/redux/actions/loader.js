import LoaderTypes from '../types/loader';

export const loaderStart = () => async dispatch => {
  dispatch({
    type: LoaderTypes.LOADER_START,
  });
};

export const loaderEnd = () => async dispatch => {
  dispatch({
    type: LoaderTypes.LOADER_END,
  });
};
