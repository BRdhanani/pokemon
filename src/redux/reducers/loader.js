import LoaderTypes from '../types/loader';

const INITIAL_STATE = {
  loading: false,
};

const loaderReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case LoaderTypes.LOADER_START:
      return {
        ...state,
        loading: true,
      };
    case LoaderTypes.LOADER_END:
      return {
        ...state,
        loading: false,
      };
    default:
      return state;
  }
};

export default loaderReducer;
