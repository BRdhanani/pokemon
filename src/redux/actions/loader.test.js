import {loaderEnd, loaderStart} from './loader';

describe('loader actions', () => {
  beforeEach(() => jest.resetAllMocks());

  it('should create LOADER_START action', () => {
    const results = loaderStart();

    expect(results).toMatchSnapshot();
  });

  it('should create loaderEnd action', () => {
    const results = loaderEnd();

    expect(results).toMatchSnapshot();
  });
});
