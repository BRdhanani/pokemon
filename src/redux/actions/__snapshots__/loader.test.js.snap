// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader actions should create LOADER_START action 1`] = `[Function]`;

exports[`loader actions should create loaderEnd action 1`] = `[Function]`;
